File: Bridge/Resources/.generated/system/iO/directoryInfo.js
    Bridge.define("System.IO.DirectoryInfo", {
        inherits: [System.IO.FileSystemInfo],
        props: {
            Exists: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Name: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Parent: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Root: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        ctors: {
            ctor: function (path) {
                System.IO.DirectoryInfo.$ctor1.call(this, path, false);
            },
            $ctor1: function (path, simpleOriginalPath) {
                this.$initialize();
                System.IO.FileSystemInfo.ctor.call(this);
            }
        },
        methods: {
            create: function () {
                throw new System.NotImplementedException();
            },
            createSubdirectory: function (path) {
                throw new System.NotImplementedException();
            },
            getFiles: function () {
                throw new System.NotImplementedException();
            },
            getFiles$1: function (searchPattern) {
                throw new System.NotImplementedException();
            },
            getDirectories: function () {
                throw new System.NotImplementedException();
            },
            getDirectories$1: function (searchPattern) {
                throw new System.NotImplementedException();
            },
            getFileSystemInfos: function () {
                throw new System.NotImplementedException();
            },
            delete: function () {
                throw new System.NotImplementedException();
            },
            delete$1: function (recursive) {
                throw new System.NotImplementedException();
            },
            moveTo: function (destDirName) {
                throw new System.NotImplementedException();
            },
            toString: function () {
                throw new System.NotImplementedException();
            }
        }
    });

File: Bridge/System/IO/DirectoryInfo.cs
using Bridge;

namespace System.IO
{

    public sealed class DirectoryInfo : FileSystemInfo
    {
        public DirectoryInfo(string path) : this(path, false)
        {
        }

        internal DirectoryInfo(string path, bool simpleOriginalPath)
        {
        }
        
        public override bool Exists
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public override string Name
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public DirectoryInfo Parent
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public DirectoryInfo Root
        {
            get
            {
                throw new NotImplementedException();
            }
        }
        
        public void Create()
        {
            throw new NotImplementedException();
        }

        public DirectoryInfo CreateSubdirectory(string path)
        {
            throw new NotImplementedException();
        }
        
        public FileInfo[] GetFiles()
        {
            throw new NotImplementedException();
        }

        public FileInfo[] GetFiles(string searchPattern)
        {
            throw new NotImplementedException();
        }

        public DirectoryInfo[] GetDirectories()
        {
            throw new NotImplementedException();
        }

        public DirectoryInfo[] GetDirectories(string searchPattern)
        {
            throw new NotImplementedException();
        }

        public FileSystemInfo[] GetFileSystemInfos()
        {
            throw new NotImplementedException();
        }
        
        public override void Delete()
        {
            throw new NotImplementedException();
        }

        public void Delete(bool recursive)
        {
            throw new NotImplementedException();
        }

        public void MoveTo(string destDirName)
        {
            throw new NotImplementedException();
        }

        public override string ToString()
        {
            throw new NotImplementedException();
        }
    }
}